### ISSUE TITLE
404 Pages

### ISSUE THREAD

**Issue by danielcj2** [bluewave-labs/bluewave-uptime#398]
I think it would be nice to have some proper 404 pages set up. Displaying `NotFoundPage` on a blank screen is a little dull. 

Ideally we'd have more than one design but for now one should do. 
- Standard 404 page that covers all general cases.
- Specific page for when a monitor isn't found. (in the future, perhaps)

**Comment by MuhammadKhalilzadeh** [comment]
@gorkemcetin I'm interested in this.
Are you going to design anything specific for this? or I can design something on my own?

**Comment by gorkem-bwl** [comment]
Sure, let me come up with designs here. What are some of the pages you envisage @MuhammadKhalilzadeh ?

**Comment by MuhammadKhalilzadeh** [comment]
What I'm suggesting is to have a single dynamic page to cover all the `HTTP Status codes`, and only differ in the font color, title, description (which are dynamic parts) according to the status code. Maybe like this:

```
1XX Blue theme
2XX Green Theme
3XX Purple Theme
4XX Yellow Theme
5XX Red Theme

```
@gorkem-bwl What do you think?

**Comment by ajhollid** [comment]
This sounds like a good approach.

It would be nice to have this page take some props too so we can customize the message and perhaps icon?

I imagine it might take `statusCode`, `msg`, and `img` perhaps?

All optional and with default values?

**Comment by MuhammadKhalilzadeh** [comment]
Yes. And also I believe it would be nice to have a `bgColor` and a `fontColor` too, so that be can make the theme fit the `statusCode`

**Comment by ajhollid** [comment]
I don't think you'll need a separate prop for that, you can just have a lookup for each status code:

```
const lookup = {
   400: { bgColor: 'red', fontColor: 'white'}
   200: { bgColor: 'green, fontColo: 'black'}
   }
```

something like that.

We could also include a catch all for all code ranges that don't have a specific code, ie all 400 errors that don't have a specific code get x style, all 200 codes that don't have a specific code get y style.

That would be a nice flexible and reusable component

**Comment by gorkem-bwl** [comment]
Heh I am a bit lost here. Let's see what you have Mo, and then we move from that point fwd :)

**Comment by MuhammadKhalilzadeh** [comment]
😂 Then, let me design something and get back to you guys

**Comment by MuhammadKhalilzadeh** [comment]
For `2XX Status Codes`

![HTTP - Status - 2XX](https://github.com/user-attachments/assets/f90307d7-c6b5-4caa-a679-8be93bd0ff0d)

**Comment by MuhammadKhalilzadeh** [comment]
For 4XX `Status Codes`

![HTTP - Status - 4XX](https://github.com/user-attachments/assets/80460f18-1b04-43b2-a3dd-cd1571fb9b64)

**Comment by MuhammadKhalilzadeh** [comment]
For 5XX Status Codes

![HTTP - Status - 5XX](https://github.com/user-attachments/assets/496cd542-4bd0-48bb-9757-393ddea93ae9)

**Comment by MuhammadKhalilzadeh** [comment]
@gorkem-bwl A page as simple as this in these different and customizable states and no need for any `imgs`.
Only props are needed:
`statusCode`, `msg` and `themeColor`.

What do you guys think?

**Comment by ajhollid** [comment]
Concept is good, but I don't think the design matches the rest of our designs really 🤔

Perhaps @gorkem-bwl can take the concept and match it to our design paradigm?

**Comment by MuhammadKhalilzadeh** [comment]
Yesss! That's exactly what I meant to do! Transferring the concept in my mind!
Yep! then let's wait for him.

**Comment by ajhollid** [comment]
> Yesss! That's exactly what I meant to do! Transferring the concept in my mind!
> Yep! then let's wait for him.

Sounds good! 

Also 200 status codes won't be needed, there's no case where we have a successful request but need to redirect to or render a success page I don't think.

**Comment by gorkem-bwl** [comment]
> Yesss! That's exactly what I meant to do! Transferring the concept in my mind! Yep! then let's wait for him.

Yup - will check this tomorrow. I am a bit exhausted after 9 hours of drive today 🤷‍♂️

**Comment by danielcj2** [comment]
I think we're getting ahead of ourselves a little. Right now, there is no need for this level of modularity. It seems to me that we're overcomplicating what appears to be an easy solution. Worrying about status codes and incorporating them in the component is irrelevant. What we really need is page to fall back on if a user enters a random route that isn't defined in `App.jsx`, which we already have in place. The other case where we would need to render a "not found" page is within the monitor parent route, specifically when a user enters a random `_id` that does not exist in the database. 

TL;DR: We need a `NotFound` page design ready to render when the user navigates to an undefined route or a non-existing monitor.

**Comment by ajhollid** [comment]
Sounds reasonable to me, I'll defer to the FE guys on FE matters as per usual 👌

**Comment by Skorpios604** [comment]
I think the idea @MuhammadKhalilzadeh is coming up with is different and I haven't really seen it in the wild at all. I agree with @danielcj2 about keeping it plain and simple and direct. We want something that's functional that lets the user know that they cannot access that route. We do not need to overwhelm them with many status codes that are irrelevant to them anyway.

**Comment by MuhammadKhalilzadeh** [comment]
You are presenting a very complex analysis of my words and I feel that you have not read all the messages or maybe I wasn't able to convey what I mean correctly. Here's what I mean:

- This is simply a concept design and not to implement!

- This is gonna be as simple as a page with a text for `statusCode` and `msg` as props like any other page we created before and gets props

Now whatever you want the page look like I'm just simply ready to code it and get the task complete. Its a nice idea to have a simple page to show the http error anyway!

**Comment by ajhollid** [comment]
Alrigh team, before we get bogged down implementation details let's wait until we have some design specs.

Once we know what the page is going to look like the implementation will be obvious as this is a simple page.  Until then let's all sit tight :smile:

**Comment by MuhammadKhalilzadeh** [comment]
😂😂😅😂😜
Cheers to everyone then 🍻

**Comment by gorkem-bwl** [comment]
My suggestion would be to go with a very simple page here. How about the one I have on Figma? Copying it here for reference:

<img width="1019" alt="Screenshot 2024-07-22 at 2 09 54 PM" src="https://github.com/user-attachments/assets/97b16f0b-c920-4cfa-a23d-b968e418f8ac">

@MuhammadKhalilzadeh if you are comfortable with this one, let's move forward :)
